feat(many): mark swrast as deprecated and replace everywhere (#75)
anbox-cloud/discourse-docs: reference/platforms.md

|----------	|-----------------------------------------------------------------------------------------------------------------------------------------------------	|
| `null`   	|  A headless-gl platform. No rendering is performed. No audio input/output. Useful for functional tests. It's used by default if no platform is specified when launching a container.                                                                       	|
| `webrtc` 	| Full-featured WebRTC based streaming platform. Includes driver and integration for AMD and NVidia GPUs as well as LLVMPipe based software rendering if no GPU is detected.  Support audio input/output. |
| `swrast` 	| (DEPRECATED) **S**oft**w**are **Rast**erization platform. A LLVMPipe based software rendering platform. Useful for visual tests. No audio input/output.                                                               	|

## Using platforms

Instructing a container to use a platform is done through the `--platform` (or `-p`) flag when launching a container, e.g.

    amc launch -p webrtc <application>

### `swrast` platform
#### Display Settings Configuration

> [note type="caution" status="Warning"]The `swrast` platform is deprecated and replaced with the `webrtc` platform starting with Anbox Cloud 1.13. You can still explicitly specify `swrast` as platform name but internally it will be mapped to the `webrtc` platform. The `webrtc` platform provides backward compatiblity with the display settings described below.[/note]

Reviewer: ```suggestion
[note type="caution" status="Warning"]The `swrast` platform is deprecated and has been replaced with the `webrtc` platform starting with Anbox Cloud 1.13. You can still explicitly specify `swrast` as platform name, but internally, it is mapped to the `webrtc` platform. The `webrtc` platform provides backward compatibility with the display settings described below.[/note]
```

If you want to change the display settings of Android container, you need to provide a combination of a numeric formatting string as follows:

>     <Display width>,<Display height>,<FPS>,<Display Density>

Reviewer: Tiny nitpick:

```suggestion
    <Display width>,<Display height>,<FPS>,<Display density>
```

Display settings for the `null` can be configured in the same way as for the `swrast` platform.

Instead of supplying the display settings via userdata through the `amc launch` command they can be alternatively written before the start of the Anbox runtime (e.g. in a `pre-start` hook) to `/var/lib/anbox/display_settings`. The format remains the same as when supplied as userdata.

### `webrtc` platform

> The `webrtc` platform is the default platform used by Anbox to provide graphical output. It supports all GPUs supported by Anbox Cloud in addition to software rendering. It is used by default when a container is launched with graphics enabled (e.g. `amc launch --enable-graphics ...` or via the Anbox Stream Gateway).

Reviewer: Let's use Enzo's suggestion.

Reviewer: The one that gets rid of "default":
```suggestion
The `webrtc` platform is used by Anbox to provide graphical output. It supports all GPUs supported by Anbox Cloud in addition to software rendering. It is used when a container is launched with `--enabled-graphics`, or via the Anbox Stream Gateway.
```

#### Configuration

The `webrtc` platform can be configured through user data provided to the container in JSON format. AMS puts the configuration data at `/var/lib/anbox/userdata`.

Field name | Type | Default | Description
-----------|------|---------|------------
`display_width` | `int` | `1280` | Width of the display provided to Android.
`display_height` | `int` | `720` | Height of the display provided to Android.
`display_density` | `int` | `240` | Density of the display provided to Android.
`fps` | `int` | `60` | Refresh rate of the display provided to Android.

> To configure the platform for example for a 1080p and 60 FPS you can set user data for a container like this:

Reviewer: ```suggestion
For example, to configure the platform for a display height of 1080p and 60 FPS, set the user data for a container like this:
```

### `webrtc` platform

> The `webrtc` platform is the default platform used by Anbox to provide graphical output. It supports all GPUs supported by Anbox Cloud next to software rendering. It is used by default when a container is launched with enabled graphics (e.g. `amc launch --enable-graphics ...` or via the Anbox Stream Gateway)

Reviewer: ```suggestion
The `webrtc` platform is the default platform used by Anbox to provide graphical output. It supports all GPUs supported by Anbox Cloud in addition to software rendering. It is used by default when a container is launched with graphics enabled (e.g. `amc launch --enable-graphics ...` or via the Anbox Stream Gateway).
```

> The `webrtc` platform is the default platform used by Anbox to provide graphical output. It supports all GPUs supported by Anbox Cloud next to software rendering. It is used by default when a container is launched with enabled graphics (e.g. `amc launch --enable-graphics ...` or via the Anbox Stream Gateway)

Reviewer 2: I would be careful with the word `default` as it could be understood as `webrtc is the default platform.`
Maybe something like
```suggestion
The `webrtc` platform is used by Anbox to provide graphical output. It supports all GPUs supported by Anbox Cloud next to software rendering. It is used when a container is launched with `--enabled-graphics`, or via the Anbox Stream Gateway.
```

Reviewer: I agree, let's use the suggestion.

#### Configuration

The `webrtc` platform can be configured through user data provided to the container which will be placed by AMS at the location `/var/lib/anbox/userdata`. The format of the configuration is JSON.

Field Name | Type | Default | Description
-----------|------|---------|------------

> `id` | `string` | `""` | ID of the streaming session
> `url` | `string` | `""` | WebRTC signaling endpoint URL
> `auth_token` | `string` | `""` | Authentication token used to talk to the signaling endpoint
> `use_insecure_tls` | `boolean` | `false` | Allow the use of insecure TLS for the WebRTC signaling endpoint
> `display_width` | `int` | `1280` | Width of the display provided to Android
> `display_height` | `int` | `720` | Height of the display provided to Android
> `display_density` | `int` | `240` | Density of the display provided to Android
> `fps` | `int` | `60` | Refresh rate of the display provided to Android
> `session` | `object` | `{}` | JSON object describing details of the streaming session
> `session.should_keep_alive` | `boolean` | `false` | Indicates wether a streaming session should be kept alive after all users have been disconnected. If set to `false` the session and the container will be stopped as soon as the last user disconnected.
> `session.max_idle_time` | `int` | `-1` | Time in minutes telling the platform how long to stay idle and alive after the last user has disconnected. If set to `-1` the platform stay alive indefinitely.

Reviewer: ```suggestion
`id` | `string` | `""` | ID of the streaming session.
`url` | `string` | `""` | WebRTC signaling endpoint URL.
`auth_token` | `string` | `""` | Authentication token used to talk to the signaling endpoint.
`use_insecure_tls` | `boolean` | `false` | Allow the use of insecure TLS for the WebRTC signaling endpoint.
`display_width` | `int` | `1280` | Width of the display provided to Android.
`display_height` | `int` | `720` | Height of the display provided to Android.
`display_density` | `int` | `240` | Density of the display provided to Android.
`fps` | `int` | `60` | Refresh rate of the display provided to Android.
`session` | `object` | `{}` | JSON object describing details of the streaming session.
`session.should_keep_alive` | `boolean` | `false` | Indicates whether a streaming session should be kept alive after all users have been disconnected. If set to `false`, the session and the container will be stopped as soon as the last user disconnected.
`session.max_idle_time` | `int` | `-1` | Time in minutes telling the platform how long to stay idle and alive after the last user has disconnected. If set to `-1`, the platform stays alive indefinitely.
```
